src/esbuild_plugins/islands.ts

```diff
@@ -0,0 +1,38 @@
+import type { Plugin } from "https://deno.land/x/esbuild@v0.17.16/mod.js";
+import { compile } from "npm:svelte/compiler";
+
+const filter = /\.svelte$/;
+const name = "mononykus/svelte-islands";
+
+export const island_wrapper = (mode: "ssr" | "dom", dir: string): Plugin => ({
+	name,
+	setup(build) {
+		build.onLoad({ filter }, async ({ path }) => {
+			const filename = path.split(dir).at(-1) ?? "Undefined.svelte";
+			const source = await Deno.readTextFile(path);
+			const { js: { code } } = compile(source, {
+				generate: mode,
+				css: "external",
```

Comment:
```suggestion
				css: "injected",
```
should the JS handle adding styles? do we want css files output?

Reply:
This has the correct behaviour in SSR components already. We don't want islands to have code about CSS.

Reply:
what if an island conditionally styles something?

Reply:
Styles are built statically, that’s part of the beauty of Svelte!

> This is all possible because your CSS is parsed (with [css-tree](https://github.com/csstree/csstree)) and statically analysed in the context of your markup. Static analysis opens the doors to all kinds of exciting future possibilities — smarter optimisations, a11y hints — that are much harder if your styles are computed dynamically at runtime. We're just getting started.

https://svelte.dev/blog/the-zen-of-just-writing-css

If you actually need dynamic styles, then they are handled inline by your JS, but you don’t have to worry about that either.

https://geoffrich.net/posts/style-directives/